src/Sylius/Bundle/CoreBundle/EventListener/TaxonDeletionListener.php

```diff
@@ -57,6 +63,35 @@ public function protectFromRemovingMenuTaxon(GenericEvent $event): void
         }
     }
 
+    public function protectFromRemovingProductMainTaxon(GenericEvent $event): void
```

Comment:
This class becoming more and more like a controller. We should split it into separate parts, but that may be a case for separate PR.

Reply:
agreed